### PR TITLE
Replaced OpenMP offloading with SYCL in DR

### PR DESCRIPTION
**Pull request by ravil-mobile**
I noticed that the GPU solution of the TPV16 scenario overflows (with respect to `P_n` variable ) when SeisSol configured with `single` precision. Therefore, the CI tests the scenario with `double` precision.

### CODE REVIEW

**Review by Thomas-Ulrich — APPROVED** [review]
Obviously, I don't understand the details, but I've read the code, and it seems to make sense (also I tested it on heisenbug and it works).
You can decide if you want to wait for a review from TUM or merge right away.

**Comment by ravil-mobile** [comment]
> Obviously, I don't understand the details, but I've read the code, and it seems to make sense (also I tested it on heisenbug and it works). You can decide if you want to wait for a review from TUM or merge right away.

@Thomas-Ulrich , thanks for a quick review. I need to fix the GPU CI pipeline first and also update the documentation because SYCL is going to be a hard-dependency for SeisSol-GPU.

**Comment by ravil-mobile** [comment]
@sebwolf-de , Can I ask you to have a look at this PR?

File: Documentation/compilation.rst
Comment (sebwolf-de) [inline Documentation/compilation.rst:143]: Nice :+1:

File: src/Numerical_aux/GaussianNucleationFunction.h
Comment (sebwolf-de) [inline src/Numerical_aux/GaussianNucleationFunction.h:12]: Please make this a template option.

File: Documentation/sycl.rst
Comment (ravil-mobile) [reply Documentation/sycl.rst:89]: No, It is still under development

File: gitlab-gpu-pipeline.yml
Comment (ravil-mobile) [reply gitlab-gpu-pipeline.yml:188]: Energy output is not ready. `Receivers` output - I forgot

File: src/Numerical_aux/GaussianNucleationFunction.h
Comment (ravil-mobile) [reply src/Numerical_aux/GaussianNucleationFunction.h]: done

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h]: done

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h]: done

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h]: done with templates. However, `startIndex` is required

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h]: done

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h]: done with templates

File: gitlab-gpu-pipeline.yml
Comment (ravil-mobile) [reply gitlab-gpu-pipeline.yml:188]: added `Receivers output` check

File: Documentation/sycl.rst
Comment (ravil-mobile) [reply Documentation/sycl.rst]: done

File: Documentation/sycl.rst
Comment (ravil-mobile) [reply Documentation/sycl.rst]: done

File: Documentation/sycl.rst
Comment (ravil-mobile) [reply Documentation/sycl.rst]: done

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (sebwolf-de) [inline src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:26]: class enum?

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (sebwolf-de) [inline src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:125]: This could be integrated in the for loop:
` for (auto index = startLoopIndex + Range::start; index < startLoopIndex + Range::end, index += Range::step)`

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (sebwolf-de) [inline src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:217]: Also consider to put this into the loop header.

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (sebwolf-de) [inline src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:252]: Also consider to put this into the loop header.

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (sebwolf-de) [inline src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:305]: Also consider to put this into the loop header.

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (sebwolf-de) [inline src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:335]: Also consider to put this into the loop header.

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (sebwolf-de) [inline src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:361]: Also consider to put this into the loop header.

File: src/Numerical_aux/Functions.h
Comment (sebwolf-de) [inline src/Numerical_aux/Functions.h:46]: Probably, we need a `#pragma omp declare simd` here to enable vectorization.

File: src/Numerical_aux/GaussianNucleationFunction.h
Comment (sebwolf-de) [inline src/Numerical_aux/GaussianNucleationFunction.h:4]: IMHO there is no need to include this here. Should rather be included in the Dynamic Rupture classes.

**Review by sebwolf-de — CHANGES_REQUESTED** [review]
Thanks, it looks way better now. Still I see some places, which are not perfect yet.

File: src/Numerical_aux/GaussianNucleationFunction.h
Comment (ravil-mobile) [reply src/Numerical_aux/GaussianNucleationFunction.h:4]: discussed

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:361]: discussed

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:305]: discussed

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:252]: discussed

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:217]: discussed

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:125]: discussed

File: src/Numerical_aux/Functions.h
Comment (ravil-mobile) [reply src/Numerical_aux/Functions.h:46]: It seems to me that there is not way to add `#pragma omp declare simd` to a templated function

File: src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h
Comment (ravil-mobile) [reply src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h:26]: done

File: src/Numerical_aux/Functions.h
Comment (ravil-mobile) [reply src/Numerical_aux/Functions.h:46]: Here is the code from the master branch

**Review by sebwolf-de — APPROVED** [review]

**Pull request merged**
